### DISCUSSION TITLE
Is JS Adapter v17 compatible with server <17?

### DISCUSSION THREAD

**Discussion by pauleustice** [keycloak/keycloak#10269] [Q&A] [answered]
Hello everyone. One of our repositories has the keycloak-js dependency at `15.1.1` and we are looking to upgrade to `17.0.0`.

Due to the keycloak monorepo and shared versioning, the release notes (AFAICT) are consolidated and as such it's not clear if there are breaking changes. The [release notes for v16](https://www.keycloak.org/docs/latest/release_notes/index.html#upgrade-to-wildfly-25-0-1) state "This requires significant changes to how Keycloak is configured" but it's not entirely clear / definitive whether this pertains only to upgrading keycloak server.

Are we able to upgrade [keycloak-js](https://www.npmjs.com/package/keycloak-js) without any other changes?

**Comment by stianst** [comment, marked as answer]
We don't test the newer adapter with older server versions, but it should work.

**Reply by pauleustice** [reply]
Thanks for the clarification 👍
